master/internal/api_checkpoint.go

```diff
@@ -327,9 +327,47 @@ func (a *apiServer) CheckpointsRemoveFiles(
 		return nil, err
 	}
 	if len(registeredCheckpointUUIDs) > 0 {
+		modelIDs := make([]int, len(registeredCheckpointUUIDs))
+		i := 0
+		for _, v := range registeredCheckpointUUIDs {
+			modelIDs[i] = v.ID
+			i++
+		}
+		var models []struct {
+			ID int
+		}
+		modelQuery := internaldb.Bun().
+			NewSelect().
+			Model(&models).
+			Table("models").
+			ColumnExpr("id").
+			Where("id IN (?)", bun.In(modelIDs))
+		if modelQuery, err = modelauth.AuthZProvider.Get().
+			FilterReadableModelsQuery(ctx, *curUser, modelQuery); err != nil {
+			return nil, err
+		}
+		err = modelQuery.Scan(ctx)
+		if err != nil {
+			return nil, err
+		}
+		accessibleModels := make(map[int]bool, len(models))
+		for _, v := range models {
+			accessibleModels[v.ID] = true
+		}
+		checkpointMsgs := make([]string, len(registeredCheckpointUUIDs))
+		i = 0
+		for k, v := range registeredCheckpointUUIDs {
+			if _, ok := accessibleModels[v.ID]; ok {
+				checkpointMsgs[i] = fmt.Sprintf("%v, registered to %v (model #%d), version %d", k, v.Name, v.ID, v.Version)
+			} else {
+				checkpointMsgs[i] = fmt.Sprintf("%v, registered to an unknown model", k)
+			}
+			i++
+		}
```

Comment:
this is very strange go style, a micro-optimization not worth the cost of readability. can we write this instead:
```suggestion
		}
		var checkpointMsgs []string
		for k, v := range registeredCheckpointUUIDs {
			if _, ok := accessibleModels[v.ID]; ok {
				msg := fmt.Sprintf("%v, registered to %v (model #%d), version %d", k, v.Name, v.ID, v.Version)
				checkpointMsgs = append(checkpointMsgs, msg)
			} else {
				msg := fmt.Sprintf("%v, registered to an unknown model", k)
				checkpointMsgs = append(checkpointMsgs, msg)
			}
		}
```

Reply:
`var checkpointMsgs []string` part can be `checkpointMsgs := make([]string, 0, len(registeredCheckpointUUIDs))`? i think thats what Ashton meant. then we can avoid using index access and also no unnecessary memory allocation i guess

Reply:
yeah, that works too. but still is a micro-optimization. preallocating it with cap `len(registeredCheckpointUUIDs)` vs just `var x []T` doesn't matter unless you've benchmarked that it does. i'm sure it is nothing compared to the sql query above it.